File: 649-dota2-senate/dota2-senate.cpp
class Solution {
public:
    string predictPartyVictory(string senate) {
        int n = senate.length();
        queue <int> rad, dire;

        for (int i = 0; i < n; i++){
            if (senate[i] == 'R'){
                rad.push(i);
            }
            else {
                dire.push(i);
            }
        }

        while (!rad.empty() && !dire.empty()){
            if (rad.front() < dire.front()){
                rad.push(n++);
            }
            else{
                dire.push(n++);
            }
            rad.pop();
            dire.pop();

        }

        if (rad.empty()){
            return "Dire";
        }
        else{
            return "Radiant";
        }
        
    }
};

File: 649-dota2-senate/dota2-senate.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        rad = []
        dire = []
        n = len(senate)

        for i in range(len(senate)):
            if senate[i] == "R":
                rad.append(i)
            else:
                dire.append(i)

        while rad and dire:
            if rad[0] < dire[0]:
                rad.append(n)
                n+=1
            else:
                dire.append(n)
                n+=1
          
            rad.pop(0)
            dire.pop(0)

        if rad:
            return "Radiant"
        if dire:
            return "Dire"

                
        
    